docs/src/main/asciidoc/async-message-passing.adoc

```diff
@@ -0,0 +1,290 @@
+
+== Installing
+
+This mechanism uses the Vert.x EventBus, so you need to enable the `vertx` extension to use this feature.
+If you are creating a new project, set the `extensions` parameter are follows:
+
+[source, subs=attributes+]
+----
+mvn io.quarkus:quarkus-maven-plugin:{quarkus-version}:create \
+    -DprojectGroupId=org.acme \
+    -DprojectArtifactId=vertx-quickstart \
+    -Dextensions="vertx"
+----
+
+If you have an already created project, the `vertx` extension can be added to an existing {project-name} project with
+the `add-extension` command:
+
+[source]
+----
+mvn quarkus:add-extension -Dextensions="vertx"
+----
+
+Otherwise, you can manually add this to the dependencies section of your `pom.xml` file:
+
+[source,xml]
+----
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-vertx</artifactId>
+</dependency>
+----
+
+== Consuming events
+
+To consume events, use the `io.quarkus.vertx.ConsumeEvent` annotation:
+
+[source, java]
+----
+package org.acme.vertx;
+
+import io.quarkus.vertx.ConsumeEvent;
+
+import javax.enterprise.context.ApplicationScoped;
+
+@ApplicationScoped
+public class GreetingService {
+
+    @ConsumeEvent                           // <1>
+    public String consume(String name) {    // <2>
+        return name.toUpperCase();
+    }
+}
+----
+<1> If not set, the address is the fully qualified name of the bean, for instance, in this snippet it's `org.acme.NameService`
+<2> The method parameter is the message body. If the method returns _something_ it's the message response.
+
+// TODO Revisit this once the blocking PR is integrated - https://github.com/quarkusio/quarkus/pull/1646
+IMPORTANT: the code consuming the event must be _non-blocking_, as it's called on the Vert.x event loop.
+
+=== Configuring the address
+
+The `@ConsumeEvent` annotation can be configured to set the address:
+
+[source, java]
+----
+@ConsumeEvent("greeting")               // <1>
+public String consume(String name) {
+    return name.toUpperCase();
+}
+----
+<1> Receive the messages sent on the `greeting` address
+
+=== Consuming without replying
+
+In the previous examples, the consumer returns a result, but it's not necessary:
+
+[source,java]
+----
+@ConsumeEvent("greeting")
+public void consume(String event) {
+    // Do something with the event
+}
+----
+
+=== Replying
+
+The _return_ of a method annotated with `@ConsumeEvent` is used as response to the incoming message.
+For instance, in the following snippet, the returned `String` is the response.
+
+[source, java]
+----
+@ConsumeEvent("greeting")
+public String consume(String name) {
+    return name.toUpperCase();
+}
+----
+
+You can also return a `CompletionStage<T>` to handle asynchronous reply:
+
+[source, java]
+----
+@ConsumeEvent("greeting")
+public CompletionStage<String> consume2(String name) {
+    return CompletableFuture.supplyAsync(name::toUpperCase);
```

Comment:
Oh sure, great, start telling people to run shit in the JDK's default executor thread pool :(

Reply:
Changed